pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -703,7 +703,9 @@ public synchronized PinotResourceManagerResponse deleteSegments(String tableName
       Preconditions.checkArgument(TableNameBuilder.isTableResource(tableNameWithType),
           "Table name: %s is not a valid table name with type suffix", tableNameWithType);
       HelixHelper.removeSegmentsFromIdealState(_helixZkManager, tableNameWithType, segmentNames);
-      _segmentDeletionManager.deleteSegments(tableNameWithType, segmentNames);
+      TableConfig tableConfig = ZKMetadataProvider.getTableConfig(_propertyStore, tableNameWithType);
+      long retentionMs = _segmentDeletionManager.getRetentionFromTableConfig(tableConfig);
+      _segmentDeletionManager.deleteSegments(tableNameWithType, segmentNames, retentionMs);
```

Comment:
Let's just pass in the `tableConfig` to this method. Calling `_segmentDeletionManager.getRetentionFromTableConfig(tableConfig)` then passing it in is slightly confusing

Reply:
IMO passing in tableConfig will be more confusing because, it is possible to use a retentionMs that is not coming from tableConfig; and also there's no point passing in table name separately either. 

Let's not restrict the API to only work on tableConfig. segmentDeletionManager only needs a zkpath and deep store path to work IMO.